config/listenaddress/listenaddress.go

```diff
@@ -42,59 +45,60 @@ const (
 
 // Configuration is the configuration for resolving a listen address.
 type Configuration struct {
-	// Hostname is the hostname to use
-	Hostname string `yaml:"hostname" validate:"nonzero"`
-
-	// Port specifies the port to listen on
-	Port *Port `yaml:"port" validate:"nonzero"`
-}
-
-// Port specifies the port to use
-type Port struct {
 	// PortType is the port type for the port
 	PortType Resolver `yaml:"portType" validate:"nonzero"`
 
-	// Value is the config specified port if using config port type.
-	Value *int `yaml:"value"`
+	// Value is the config specified listen address if using config port type.
+	Value *string `yaml:"value"`
 
-	// EnvVarName is the environment specified port if using environment port type.
-	EnvVarName *string `yaml:"envVarName"`
+	// EnvListenPort specifies the environment variable name for the listen address port.
+	EnvListenPort *string `yaml:"envListenPort"`
+
+	// EnvListenPort specifies the environment variable name for the listen address hostname.
+	EnvListenHost *string `yaml:"envListenHost"`
```

Comment:
Maybe EnvVarListenHost and EnvVarListenPort to match the host ID resolver naming of envVarXYZ?